com.unity.uiwidgets/Runtime/engine/UIWidgetsExternalTextureRegistry.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using Unity.UIWidgets.foundation;
+
+namespace Unity.UIWidgets.engine {
+    public partial class UIWidgetsPanelWrapper {
+        readonly Dictionary<IntPtr, int> externalTextures = new Dictionary<IntPtr, int>();
+
+        void ReleaseExternalTextures() {
+            foreach(var texturePair in externalTextures) {
+                var internalTextureId = texturePair.Value;
+                unregisterTexture(internalTextureId);
+            }
+            
+            externalTextures.Clear();
+        }
+
+        void ReleaseExternalTexture(IntPtr externalTexturePtr) {
+            if (externalTextures.ContainsKey(externalTexturePtr)) {
+                var internalTextureId = externalTextures[externalTexturePtr];
+                unregisterTexture(internalTextureId);
```

Comment:
`ReleaseExternalTexture` is only used at public void ReleaseExternalTexture(int internalTextureId). Is it necessary to separate like this?  I prefer write like this:
```
public void ReleaseExternalTexture(int internalTextureId) {
            var externalTexturePtr = IntPtr.Zero;
            foreach(var texturePair in externalTextures) {
                var curInternalTextureId = texturePair.Value;
                if (curInternalTextureId == internalTextureId) {
                    D.assert(externalTexturePtr == IntPtr.Zero, () => $"The internal texture id {internalTextureId} is shared by different external textures {externalTexturePtr} and {curInternalTextureId}");
                    externalTexturePtr = texturePair.Key;
                    unregisterTexture(curInternalTextureId);
                    externalTextures.Remove(externalTexturePtr);
                }
            }
        }
```
This will omit one exist check.